rocketpy/environment/environment.py

```diff
@@ -372,44 +446,177 @@ def __init__(
+        self.pressure = Function(
+            source,
+            inputs="Height Above Sea Level (m)",
+            outputs="Pressure (Pa)",
+            interpolation="linear",
+        )
 
-        # Recalculate Earth Radius (meters)
-        self.earth_radius = self.calculate_earth_radius(
-            lat=self.latitude,
-            semi_major_axis=self.ellipsoid.semi_major_axis,
-            flattening=self.ellipsoid.flattening,
+    def __set_barometric_height_function(self, source):
+        self.barometric_height = Function(
+            source,
+            inputs="Pressure (Pa)",
+            outputs="Height Above Sea Level (m)",
+            interpolation="linear",
+            extrapolation="natural",
+        )
+
+    def __set_temperature_function(self, source):
+        self.temperature = Function(
+            source,
+            inputs="Height Above Sea Level (m)",
+            outputs="Temperature (K)",
+            interpolation="linear",
+        )
+
+    def __set_wind_velocity_x_function(self, source):
+        self.wind_velocity_x = Function(
+            source,
+            inputs="Height Above Sea Level (m)",
+            outputs="Wind Velocity X (m/s)",
+            interpolation="linear",
+        )
+
+    def __set_wind_velocity_y_function(self, source):
+        self.wind_velocity_y = Function(
+            source,
+            inputs="Height Above Sea Level (m)",
+            outputs="Wind Velocity Y (m/s)",
+            interpolation="linear",
+        )
+
+    def __set_wind_speed_function(self, source):
+        self.wind_speed = Function(
+            source,
+            inputs="Height Above Sea Level (m)",
+            outputs="Wind Speed (m/s)",
+            interpolation="linear",
+        )
+
+    def __set_wind_direction_function(self, source):
+        self.wind_direction = Function(
+            source,
+            inputs="Height Above Sea Level (m)",
+            outputs="Wind Direction (Deg True)",
+            interpolation="linear",
+        )
+
+    def __set_wind_heading_function(self, source):
+        self.wind_heading = Function(
+            source,
+            inputs="Height Above Sea Level (m)",
+            outputs="Wind Heading (Deg True)",
+            interpolation="linear",
         )
 
+    def __reset_barometric_height_function(self):
+        # NOTE: this assumes self.pressure and max_expected_height are already set.
+        self.barometric_height = self.pressure.inverse_function()
+        if callable(self.barometric_height.source):
+            # discretize to speed up flight simulation
+            self.barometric_height.set_discrete(
+                0,
+                self.max_expected_height,
+                100,
+                extrapolation="constant",
+                mutate_self=True,
+            )
+        self.barometric_height.set_inputs("Pressure (Pa)")
+        self.barometric_height.set_outputs("Height Above Sea Level (m)")
```

Comment:
`__reset_barometric_height` uses set_discrete, but `__set_barometric_height` does not?

With the way you used it, this seems to be correct, but just checking in case this is not intended

Reply:
Ok, I see it now. I will make sure the `__set_barometric_height` also discretizes the Function if needed.